File: api/internal/models/wishlist.go
package models

import "time"

type WishlistItem struct {
	ID        string    `json:"id"`
	HouseID   string    `json:"house_id"`
	UserID    string    `json:"user_id"`
	UserName  string    `json:"user_name"`
	URL       string    `json:"url"`
	Title     string    `json:"title"`
	ImageURL  string    `json:"image_url"`
	CreatedAt time.Time `json:"created_at"`
}

File: web/src/services/wishlist.ts
import { api } from "./api"

export interface WishlistItem {
  id: string
  house_id: string
  user_id: string
  user_name: string
  url: string
  title: string
  image_url: string
  created_at: string
}

export async function getWishlist(): Promise<WishlistItem[]> {
  const response = await api.get<WishlistItem[]>("/wishlist")
  return response.data
}

export async function addWishlistItem(url: string): Promise<WishlistItem> {
  const response = await api.post<WishlistItem>("/wishlist", { url })
  return response.data
}

export async function deleteWishlistItem(id: string): Promise<void> {
  await api.delete("/wishlist", { data: { id } })
}
